src/policy_engine/opa/mod.rs

```diff
@@ -108,21 +135,42 @@ mod tests {
         .to_string()
     }
 
-    #[test]
-    fn test_evaluate() {
+    #[tokio::test]
+    async fn test_evaluate() {
         let opa = OPA {
-            policy_file_path: PathBuf::from("./src/policy_engine/opa/default_policy.rego"),
+            policy_dir_path: PathBuf::from("./src/policy_engine/opa"),
         };
+        let default_policy_id = "default_policy".to_string();
 
         let reference_data: HashMap<String, Vec<String>> =
             serde_json::from_str(&dummy_reference(5)).unwrap();
 
-        let res = opa.evaluate(reference_data.clone(), dummy_input(5, 5));
+        let res = opa
+            .evaluate(
+                reference_data.clone(),
+                dummy_input(5, 5),
+                Some(default_policy_id.clone()),
+            )
+            .await;
         assert!(res.is_ok(), "OPA execution() should be success");
         assert!(res.unwrap().0 == true, "allow should be true");
 
-        let res = opa.evaluate(reference_data, dummy_input(0, 0));
+        let res = opa
+            .evaluate(reference_data, dummy_input(0, 0), Some(default_policy_id))
+            .await;
         assert!(res.is_ok(), "OPA execution() should be success");
         assert!(res.unwrap().0 == false, "allow should be false");
     }
+
+    #[tokio::test]
+    async fn test_set_policy() {
+        let mut opa = OPA::new(PathBuf::from("./test_data")).unwrap();
+        let input = SetPolicyInput {
+            r#type: "opa".to_string(),
+            policy_id: "test".to_string(),
+            policy: "cGFja2FnZSBwb2xpY3kKCmltcG9ydCBmdXR1cmUua2V5d29yZHMuZXZlcnkKCmRlZmF1bHQgYWxsb3cgPSBmYWxzZQo=".to_string()
```

Comment:
Could you show the plaintext of the policy here and encode it into base64? This might help to understand the policy